### PR TITLE
[WIP] Project labels are per repo

### PR DESCRIPTION
**Pull request by ibawt**

### CODE REVIEW

**Comment by qq99** [comment]
sweet!

**Comment by ibawt** [comment]
figuring out rspec and factory girl :S  I'm not too sure what I'm doing lol

**Comment by kmcphillips** [comment]
Swoon at guard/spork!

File: app/controllers/issues_controller.rb [hunk 1]
```diff
@@ -84,12 +84,7 @@ def milestone_filtered?
   end
 
   def labels_for_repo(repo)
-    #TODO make a project_labels_repository table for this
-    labels = @project.labels
-    repo_labels = @client.get("repos/#{repo.slug}/labels").collect(&:name)
-    labels = labels.select { |l| repo_labels.include?(l.name) }
-    labels = labels.collect(&:name).join(",")
-
+    labels = @project.labels.where(repository: repo).collect(&:name).join(',')
```
Comment (kmcphillips) [inline app/controllers/issues_controller.rb:87]: This should be refactored up into the project model.

File: spec/controllers/projects_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,141 @@
+require 'spec_helper'
+
+describe ProjectsController, :vcr  do
+  login_user
```
Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:4]:
``` ruby
before do
  login_user
end
```

File: spec/controllers/projects_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  let(:project) { FactoryGirl.create(:project_with_repositories) }
+
+  describe "#index" do
+    it "populates the projects array" do
+      get :index
+      assigns(:projects).should eq([project])
+    end
+    it "render the index view" do
+      get :index
+      response.should render_template :index
```
Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:15]: The `.should` syntax is slowly being deprecated. Replaced with. `expect(response).to render_template(:index)`

Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:15]: This syntax isn't wrong, just not current.

Comment (ibawt) [inline spec/controllers/projects_controller_spec.rb:15]: cool, I have no idea what I'm doing in rspec.  It took me a while just to even get what's happening.

File: spec/controllers/projects_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    it "renders the show view" do
+      get :show, id: project
+
+      response.should render_template :show
+    end
+  end
+
+  describe "#new" do
+    it "creates a new project for population" do
+      Project.should_receive(:new)
```
Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:33]:
``` ruby
expect(Project).to receive(:new)
```

File: spec/factories.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+FactoryGirl.define do
+  factory :user do |user|
```
Comment (kmcphillips) [inline spec/factories.rb:2]: I think this syntax is deprecated. I'll refactor the factories in a bit.

File: app/controllers/issues_controller.rb (hunk 1 above)
Comment (ibawt) [inline app/controllers/issues_controller.rb:87]: yah I was going to, but all this setup with VCR and stuff I wanted to get it into master asap. I'm refactoring some stuff right now with better tests.

File: app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -21,48 +21,41 @@ def change_heading
...
+      render json: { refresh: true }, :status => :locked
+    else
+      params[:issues].each_value do |issue|
+        my_issue = Issue.find(issue['id'].to_i)
+        coords = issue['coords']
+        my_issue.row = coords['row'].to_i
+        my_issue.col = coords['col'].to_i
+        my_issue.width = coords['size_x'].to_i
+        my_issue.height = coords['size_y'].to_i
+        my_issue.save
```
Comment (ibawt) [inline app/controllers/projects_controller.rb:52]: this is the part I'm refactoring as it's ugly and slow.

File: spec/controllers/projects_controller_spec.rb (hunk 2 above)
Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:4]: Oh I see it's a class macro with the before in it. KCCO.

File: spec/controllers/projects_controller_spec.rb (hunk 3 above)
Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:15]: It actually looks really good! Love rspec and these are good specs.

Comment (ibawt) [inline spec/controllers/projects_controller_spec.rb:15]: is there a better resource for Rspec?  I find I'm not familiar with all the "verbs" so I just try and get something to work or I'm following an old blog article :(

Comment (kmcphillips) [inline spec/controllers/projects_controller_spec.rb:15]: http://betterspecs.org/ is good. The expectations repo is good http://rubydoc.info/gems/rspec-expectations/frames

Honestly I've been using them for so long I don't think about it anymore.

File: spec/controllers/projects_controller_spec.rb (hunk 2 above)
Comment (ibawt) [inline spec/controllers/projects_controller_spec.rb:4]: Yah honestly, I just copied the devise readme for rspec.

**Comment by ibawt** [comment]
There's some specific VCR stuff now, it's just an ENV variable with a github token so that it can not get rate limited.  It's in the README.

**Pull request merged**